service/matching/forwarder.go

```diff
@@ -129,13 +129,14 @@ func (fwdr *Forwarder) ForwardTask(ctx context.Context, task *internalTask) erro
 		return errForwarderSlowDown
 	}
 
-	var expirationDuration time.Duration
-	expirationTime := timestamp.TimeValue(task.event.Data.ExpiryTime)
-	if !expirationTime.IsZero() {
-		expirationDuration = time.Until(expirationTime)
-		if expirationDuration <= 0 {
+	var expirationDuration *durationpb.Duration
+	expirationTime := task.event.Data.ExpiryTime.AsTime()
+	if task.event.Data.ExpiryTime != nil && !expirationTime.IsZero() {
```

Comment:
This check _is_ needed, and skipping the nil check causes a test failure because of how the zero protobuf time (not a zero go time) is handled.

we probably _only_ need to check whether ExpiryTime is nil, honestly. I'll see what happens if I do